lib/puppet/provider/cisco_vxlan_vtep_vni/nxapi.rb

```diff
@@ -0,0 +1,187 @@
+  VXLAN_VTEP_VNI_NON_BOOL_PROPS = [
+    :ingress_replication,
+    :multicast_group,
+  ]
+
+  VXLAN_VTEP_VNI_BOOL_PROPS = [
+    :suppress_arp
+  ]
+
+  VXLAN_VTEP_VNI_ALL_PROPS = VXLAN_VTEP_VNI_NON_BOOL_PROPS + VXLAN_VTEP_VNI_BOOL_PROPS
+
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@vtep_vni',
+                                            VXLAN_VTEP_VNI_NON_BOOL_PROPS)
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:bool, self, '@vtep_vni',
+                                            VXLAN_VTEP_VNI_BOOL_PROPS)
+
+  def initialize(value={})
+    super(value)
+    interface = @property_hash[:interface]
+    vni = @property_hash[:vni]
+    @vtep_vni = Cisco::VxlanVtepVni.vnis[interface][vni] unless
+      interface.nil? || vni.nil?
+    @property_flush = {}
+    debug 'Created provider instance of cisco_vxlan_vtep_vni.'
+  end
+
+  def self.properties_get(interface, vni, vni_instance)
+    debug "Checking instance, name #{interface} #{vni}"
+    current_state = {
+      name:   	  "#{interface} #{vni}",
+      interface: interface,
+      vni:       vni,
+      assoc_vrf: vni_instance.assoc_vrf,
+      ensure:    :present,
+      peer_ips:  vni_instance.peer_list,
+    }
+    # Call node_utils getter for each property
+    VXLAN_VTEP_VNI_NON_BOOL_PROPS.each do |prop|
+      current_state[prop] = vni_instance.send(prop)
+    end
+    VXLAN_VTEP_VNI_BOOL_PROPS.each do |prop|
+      val = vni_instance.send(prop)
+      if val.nil?
+        current_state[prop] = nil
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    vnis = []
+    Cisco::VxlanVtepVni.vnis.each do |interface, all_vnis|
+      all_vnis.each do |vni, vni_instance|
+        vnis << properties_get(interface, vni, vni_instance)
+      end
+    end
+    vnis
+  end
+
+  def self.prefetch(resources)
+    vnis = instances
+    resources.keys.each do |id|
+      provider = vnis.find do |vni|
+        vni.interface.to_s == resources[id][:interface].to_s &&
+        vni.vni == resources[id][:vni] &&
+        vni.assoc_vrf.to_s == resources[id][:assoc_vrf].to_s
+      end
+      resources[id].provider = provider unless provider.nil?
+    end
+  end # self.prefetch
+
+  def exists?
+    (@property_hash[:ensure] == :present)
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def instance_name
+    vni
+  end
+
+  def properties_set(new_vni=false)
+    VXLAN_VTEP_VNI_ALL_PROPS.each do |prop|
+      next unless @resource[prop]
+      send("#{prop}=", @resource[prop]) if new_vni
+      unless @property_flush[prop].nil?
+        @vtep_vni.send("#{prop}=", @property_flush[prop]) if
+          @vtep_vni.respond_to?("#{prop}=")
+      end
+    end
+    # node utils is named peer_list= instead of peer_ips=
+    return unless @resource[:peer_ips]
```

Comment:
Do we have to change something in the NU for this change? I made the change in the provider but after applying the property peer_list initially, any changes to the list in the manifest are not applied to the switch. This was working with the custom method which means NU should be fine.

Info: Applying configuration version '1453771610'
#<Cisco::VxlanVtep:0x00000002f612b8>
Notice: /Stage[main]/Ciscopuppet::Demo_vxlan/Cisco_vxlan_vtep_vni[nve1 10000]/peer_list: peer_list changed ['2.2.2.2', '1.1.1.3', '3.3.3.3'] to '4.4.4.4 2.2.2.6 3.3.3.9'
Notice: /Stage[main]/Ciscopuppet::Demo_vxlan/Cisco_vxlan_vtep_vni[nve1 20000]/peer_list: peer_list changed [] to 'default'
Notice: Applied catalog in 0.70 seconds
root@agent-lab1-nx#puppet resource cisco_vxlan_vtep_vni
cisco_vxlan_vtep_vni { 'nve1 10000':
  ensure              => 'present',
  ingress_replication => 'static',
  peer_list           => ['2.2.2.2', '1.1.1.3', '3.3.3.3'],
  suppress_arp        => 'false',
}
cisco_vxlan_vtep_vni { 'nve1 20000':
  ensure          => 'present',
  multicast_group => '224.1.1.1',
  suppress_arp    => 'false',
}


Here is the link to my latest code:
https://github.com/alok-aggarwal/cisco-network-puppet-module/tree/develop/lib